sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java

```diff
@@ -625,6 +678,37 @@ public String toString() {
         }
     }
 
+    public static class ResourcePollResult {
+        private final String status;
+
+        @JsonCreator
+        public ResourcePollResult(@JsonProperty(value = "status", required = true) String status) {
+            this.status = status;
+        }
```

Comment:
Usually we will not get `JacksonException` on de-serialize, even when the type does not match (likely result in a object with everything null).

However in this case of required property in ctor, Jackson will throw.